apisix/init.lua

```diff
@@ -266,12 +266,21 @@ local function set_upstream_host(api_ctx, picked_server)
 end
 
 
+local upstream_proxy_headers = {
+    var_x_forwarded_proto = 'X-Forwarded-Proto',
+    var_x_forwarded_for = 'X-Forwarded-For',
```

Comment:
For X-Forwarded-For, we need to append the client's real-ip to it, see https://github.com/apache/apisix/blob/a0efeca194bcca58e62d73c0a19464f480a1b7c9/t/APISIX.pm#L698

Reply:
I'm uncertain about this. `proxy-rewrite` plugin simply overwrites the original value for other headers. If `X-Forward-For` is specialized, the behavior is not consistent, which may be a confusion for the users.